homeassistant/components/lovelace/__init__.py

```diff
@@ -1,19 +1,73 @@
 """Lovelace UI."""
+import logging
+import uuid
 import voluptuous as vol
+from collections import OrderedDict
+from typing import Union, List, Dict
 
 from homeassistant.components import websocket_api
-from homeassistant.util.yaml import load_yaml
 from homeassistant.exceptions import HomeAssistantError
 
+_LOGGER = logging.getLogger(__name__)
 DOMAIN = 'lovelace'
+REQUIREMENTS = ['ruamel.yaml==0.15.72']
 
 OLD_WS_TYPE_GET_LOVELACE_UI = 'frontend/lovelace_config'
 WS_TYPE_GET_LOVELACE_UI = 'lovelace/config'
+
 SCHEMA_GET_LOVELACE_UI = websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
     vol.Required('type'): vol.Any(WS_TYPE_GET_LOVELACE_UI,
                                   OLD_WS_TYPE_GET_LOVELACE_UI),
 })
 
+JSON_TYPE = Union[List, Dict, str]  # pylint: disable=invalid-name
+
+
+def save_yaml(fname, data):
+    from ruamel.yaml import YAML
+    """Save a YAML file."""
+    yaml = YAML()
+    yaml.explicit_start = True
+    yaml.indent(sequence=4, offset=2)
+    try:
+        with open(fname, "w", encoding='utf-8') as conf_file:
+            yaml.dump(data, conf_file)
```

Comment:
This is dangerous, dump the yaml first to a string, then open file and truncate. Otherwise, if the dumping fails, the lovelace config is wiped out.

Reply:
Like this?
```
    output = StringIO()
    yaml.dump(data, output)
    print(output.getvalue())
    with open("conf.yaml", 'w', encoding='utf-8') as conf_file:
        output.seek(0)
        copyfileobj(output, conf_file)
    output.close()
```

Reply:
So I was googling it, and that's pretty inefficient according to the author of ruamel.yaml.

Why not do this

https://github.com/home-assistant/home-assistant/blob/9c52a3ce223c06f2d29b0e61c59d660dd180e5b8/homeassistant/util/json.py#L53-L56